bioptim/limits/penalty.py

```diff
@@ -1300,6 +1352,38 @@ def _add_track_data_to_plot(
             axes_idx=axes_idx,
         )
 
+    @staticmethod
+    def _get_states_or_controls_with_specified_var(pn: PenaltyNodes, s: str):
+        """
+        Extracting controls or states depending on if the selected optimisation variables s are defined as states or controls
+
+        Parameters
+        ----------
+        pn: PenaltyNodes
+                The penalty node elements
+        Returns
+        ----------
+        MX.sym
+            states or control, with the selected optimisation variables s in it.
+        scalar
+            first_idx, the index of the first component in the control or state vector with the selected optimisation variables s
+        scalar
+            idx, the position of the vector in the controls or states vector of the selected optimisation variables s
```

Comment:
Isn't `first_idx = idx[0]` ? If so, then `first_idx` is useless, otherwise, please find new names since I don't understand what they are

Reply:
if I have a two dof system, torque driven and I want the qdot. Then first_idx is 2, and idx = 1.
if it's a three dof system, first_idx is 3, and idx = 1.

Reply:
Then why idx is not 2 or (2, 3) for first and all qdot respectively? What is the information carried by idx?